### PR TITLE
Bulk cleanup of hudson.slaves and jenkins.slaves

### PR DESCRIPTION
**Pull request by oleg-nenashev**
- [x] Use Jenkins.get() instead of the deprecated getInstance()
- [x] Remove unused imports
- [x] Use diamond type definitions where possible
- [x] Squash similar extension handlers
- [x] Other minor refactorings and NPE checks

* No tests
* Changelogs are not required

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/slaves/SlaveComputer.java [hunk 1]
```diff
@@ -486,7 +488,11 @@ public String getAbsoluteRemoteFs() {
 
     static class LoadingPrefetchCacheCount extends MasterToSlaveCallable<Integer,RuntimeException> {
         @Override public Integer call() {
-            return Channel.current().classLoadingPrefetchCacheCount.get();
+            Channel c = Channel.current();
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: Leftover from #3145

File: core/src/main/java/hudson/slaves/SlaveComputer.java [hunk 2]
```diff
@@ -899,13 +905,15 @@ public static VirtualChannel getChannelToMaster() {
     /**
      * Helper method for Jelly.
      */
+    @Restricted(DoNotUse.class)
+    @RestrictedSince("TODO")
```
Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: Good idea, but I'm calling it out in this comment anyway for more visibility among all the cleanup changes.

**Review by daniel-beck — APPROVED** [review]

File: core/src/main/java/hudson/slaves/SlaveComputer.java (hunk 2 above)
Comment (oleg-nenashev) [reply core/src/main/java/hudson/slaves/SlaveComputer.java]: agreed

File: core/src/main/java/hudson/slaves/SlaveComputer.java [hunk 3]
```diff
@@ -498,7 +504,7 @@ public String getAbsoluteRemoteFs() {
         @Override public Long call() {
             Channel c = Channel.current();
             if (c == null) {
-                return Long.valueOf(-1);
+                return (long) -1;
```
Comment (Wadeck) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: why not `-1L` ?

File: core/src/main/java/hudson/slaves/WorkspaceList.java [hunk 4]
```diff
@@ -112,7 +112,9 @@ public String toString() {
         public final @Nonnull FilePath path;
 
         protected Lease(@Nonnull FilePath path) {
-            path.getRemote(); // null check
+            if (path == null) { // Protection from old API
+                throw new NullPointerException("The specified FilePath is null");
```
Comment (Wadeck) [inline core/src/main/java/hudson/slaves/WorkspaceList.java]: Personally not totally convinced that the `NPE` is the correct way to manage this case. In the core, there are lots more `IAE` (> 40) than `NPE` (~8) for such cases.

More generally speaking, there is a debate concerning the fact that only the JVM should throw `NPE` but seems that `Effective Java` says the contrary.

So -0.1 from me

File: core/src/main/java/hudson/slaves/SlaveComputer.java (hunk 3 above)
Comment (oleg-nenashev) [reply core/src/main/java/hudson/slaves/SlaveComputer.java]:
```suggestion
                return -1L;
```

**Review by Wadeck — COMMENTED** [review]
Due to my comment above, I will neither approve nor block the PR

**Comment by oleg-nenashev** [comment]
@Wadeck 

To address your outstanding comment...

> More generally speaking, there is a debate concerning the fact that only the JVM should throw NPE

[NPE Javadoc](https://docs.oracle.com/javase/8/docs/api/?java/lang/NullPointerException.html) is actually explicit about that: "Applications should throw instances of this class to indicate other illegal uses of the null object. NullPointerException objects may be constructed by the virtual machine as if suppression were disabled and/or the stack trace was not writable."

Although I admit existence of such debates, Jenkins core follows Sun/Oracle codestyle, and hence I believe it's OK to follow Javadoc guidelines as well

**Pull request merged**